handler_address.go

```diff
@@ -92,8 +117,49 @@ func deleteAddressHash(hash string, req events.APIGatewayV2HTTPRequest) *events.
 	return createOutput("ok", 200)
 }
 
+func deleteAddressHashByOrganization(hash string, organizationInfo *organizationRequestBody, req events.APIGatewayV2HTTPRequest) *events.APIGatewayV2HTTPResponse {
+	addressRepo := address.GetResolveRepository()
+	currentAddress, err := addressRepo.Get(hash)
+	if err != nil {
+		log.Print(err)
+		return createError("error while fetching address record", 500)
+	}
+
+	orgRepo := organisation.GetResolveRepository()
+	currentOrg, err := orgRepo.Get(organizationInfo.OrganizationHash)
+	if err != nil {
+		log.Print(err)
+		return createError("error while fetching organization record", 500)
+	}
+
+	if currentAddress == nil {
+		return createError("cannot find address record", 404)
+	}
+
+	if currentOrg == nil {
+		return createError("cannot find organization record", 404)
+	}
+
+	//Checks if the userhash+orghash matches the hash to be deleted
+	if !pkgAddress.VerifyHash(hash, organizationInfo.UserHash, organizationInfo.OrganizationHash) {
+		return createError("error validating address", 500)
```

Comment:
I think it should not be a 500, but a 4xx return code, as this is an issue from the client-side, not the server-side.

Reply:
You’re right. It should be a 401